proxy/logging/LogConfig.cc

```diff
@@ -765,13 +775,21 @@ LogConfig::update_space_used()
     if (sret != -1 && S_ISREG(sbuf.st_mode)) {
       total_space_used += (int64_t)sbuf.st_size;
 
-      if (auto_delete_rolled_files && LogFile::rolled_logfile(entry->d_name) && candidate_count < MAX_CANDIDATES) {
+      if (auto_delete_rolled_files && LogFile::rolled_logfile(entry->d_name)) {
         //
-        // then add this entry to the candidate list
+        // then check if the candidate belongs to any given log type
         //
-        candidates[candidate_count].name  = ats_strdup(path);
-        candidates[candidate_count].size  = (int64_t)sbuf.st_size;
-        candidates[candidate_count].mtime = sbuf.st_mtime;
+        ts::TextView type_name(entry->d_name, strlen(entry->d_name));
+        auto suffix = type_name;
+        type_name.remove_suffix(suffix.remove_prefix(suffix.find('.') + 1).remove_prefix(suffix.find('.')).size());
```

Comment:
I enhanced `TextView` to have `remove_suffix_at`. E.g.
```
type_name.remove_suffix_at('.').remove_suffix_at('.');
```

Reply:
Here I'm trying to keep the chars before the second occurrence of '.'.

Reply:
`typename.remove_suffix(suffix.remove_prefix('.').remove_prefix(.').size() + 1);`. "A.B.C" -> "A".
`typename.remove.prefix('.').remove_suffix('.'); "A.B.C" -> "B".